crates/risc0-bonsai/src/host.rs

```diff
@@ -39,242 +33,63 @@ pub struct RecoveredBonsaiSession {
-                    Err(e) => {
-                        use ::bonsai_sdk::SdkErr::*;
-                        match e {
-                            InternalServerErr(s) => {
-                                let err = format!("Got HHTP 500 from Bonsai: {}", s);
-                                warn!(err);
-                                Err(backoff::Error::transient(err))
-                            }
-                            HttpErr(e) => {
-                                let err = format!("Reconnecting to Bonsai: {}", e);
-                                error!(err);
-                                Err(backoff::Error::transient(err))
-                            }
-                            HttpHeaderErr(e) => {
-                                let err = format!("Reconnecting to Bonsai: {}", e);
-                                error!(err);
-                                Err(backoff::Error::transient(err))
-                            }
-                            e => {
-                                let err = format!("Got unrecoverable error from Bonsai: {}", e);
-                                error!(err);
-                                Err(backoff::Error::permanent(err))
-                            }
-                        }
-                    }
-                }
-            },
-        )
-    };
-}
-
 /// A [`Risc0BonsaiHost`] stores a binary to execute in the Risc0 VM and prove in the Risc0 Bonsai API.
 #[derive(Clone)]
 pub struct Risc0BonsaiHost<'a> {
     elf: &'a [u8],
     env: Vec<u8>,
     assumptions: Vec<AssumptionReceipt>,
     image_id: Digest,
-    client: Option<Client>,
-    ledger_db: LedgerDB,
+    _ledger_db: LedgerDB,
 }
 
 impl<'a> Risc0BonsaiHost<'a> {
     /// Create a new Risc0Host to prove the given binary.
-    pub fn new(elf: &'a [u8], api_url: String, api_key: String, ledger_db: LedgerDB) -> Self {
+    pub fn new(elf: &'a [u8], ledger_db: LedgerDB) -> Self {
         // Compute the image_id, then upload the ELF with the image_id as its key.
         // handle error
         let image_id = compute_image_id(elf).unwrap();
 
         tracing::trace!("Calculated image id: {:?}", image_id.as_words());
 
-        // handle error
-        let client = if !api_url.is_empty() && !api_key.is_empty() {
-            tracing::debug!("Uploading image with id: {}", image_id);
-            let elf = elf.to_vec();
-            thread::spawn(move || {
-                let client = Client::from_parts(api_url, api_key, risc0_zkvm::VERSION)
-                    .expect("Failed to create Bonsai client; qed");
-
-                client
-                    .upload_img(hex::encode(image_id).as_str(), elf)
-                    .expect("Failed to upload image; qed");
-
-                tracing::debug!("Image uploaded");
-
-                Some(client)
-            })
-            .join()
-            .unwrap()
-        } else {
-            None
-        };
+        match std::env::var("RISC0_PROVER") {
```

Comment:
nit: I would implement

```rust
enum ProverType {}

impl From<&'static str> for ProverType
```
and match prover types... cleaner.

Reply:
it's not gonna be used anywhere else so I'm not gonna skip this one